activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -366,10 +366,15 @@ def type_cast_calculated_value(value, type, operation = nil)
       end
     end
 
-    # TODO: refactor to allow non-string `select_values` (eg. Arel nodes).
     def select_for_count
       if select_values.present?
-        select_values.join(", ")
+        select_values.map do |field|
```

Comment:
Do we even need to join strings here at all? If we just wrap the array in an `Arel::Nodes::Count`, the visitor should handle this for us.